src/app/shared/directives/back-button.directive.ts

```diff
@@ -1,14 +1,18 @@
 import { Directive, HostListener } from '@angular/core';
 import { Location } from '@angular/common';
+import { ClearErrorMessage } from '@app/store/actions/Error.actions';
+import { Store } from '@ngrx/store';
+import { IAppState } from '@app/store/state/app.state';
 
 @Directive({
   selector: '[vtmBackButton]'
 })
 export class BackButtonDirective {
-  constructor(private location: Location) {}
+  constructor(private location: Location, private store: Store<IAppState>) {}
 
   @HostListener('click')
   onClick() {
+    this.store.dispatch(new ClearErrorMessage());
```

Comment:
This is not the implementation needed. Please look at the existing `error.effect.ts` and lets discuss how you will handle this